cardano-node/src/Cardano/CLI/Helpers.hs

```diff
@@ -0,0 +1,308 @@
+
+prettyAddress :: CC.Common.Address -> Text
+prettyAddress addr = TL.toStrict
+  $  F.format CC.Common.addressF         addr <> "\n"
+  <> F.format CC.Common.addressDetailedF addr
+
+prettySigningKeyPub :: SigningKey -> Text
+prettySigningKeyPub sKey  =
+  let vk = Crypto.toVerification sKey
+  in TL.toStrict
+    $  "public key hash: " <> (F.format Crypto.hashHexF . CC.Common.addressHash $ vk) <> "\n"
+    <> "     public key: " <> (Builder.toLazyText . Crypto.formatFullVerificationKey $ vk)
+
+-- TODO:  we need to support password-protected secrets.
+readSigningKey :: CLIOps IO -> SigningKeyFile -> IO SigningKey
+readSigningKey co (SigningKeyFile fp) =
+  coDeserialiseDelegateKey co fp =<< LB.readFile fp
+
+readVerificationKey :: VerificationKeyFile -> IO Crypto.VerificationKey
+readVerificationKey (VerificationKeyFile fp) = do
+  vkB <- SB.readFile fp
+  case Crypto.parseFullVerificationKey . fromString $ UTF8.toString vkB of
+    Left e -> throwIO . VerificationKeyDeserialisationFailed fp $ pack $ show e
+    Right x -> pure x
+
+
+txSpendGenesisUTxOByronPBFT :: CC.Genesis.Config -> SigningKey -> Address -> NonEmpty TxOut -> GenTx (ByronBlockOrEBB ByronConfig)
+txSpendGenesisUTxOByronPBFT gc sk genRichAddr outs =
+  let vk         = Crypto.toVerification sk
+      txattrs    = mkAttributes ()
+      tx         = UnsafeTx (pure txIn) outs txattrs
+      txIn      :: CC.UTxO.TxIn
+      txIn       = handleMissingAddr $ fst <$> Map.lookup genRichAddr initialUtxo
+      handleMissingAddr :: Maybe CC.UTxO.TxIn -> CC.UTxO.TxIn
+      handleMissingAddr  = fromMaybe . error
+        $  "\nGenesis richmen UTxO has no address\n"
+        <> (unpack $ prettyAddress genRichAddr)
+        <> "\n\nIt has the following, though:\n\n"
+        <> Cardano.Prelude.concat (unpack . prettyAddress <$> Map.keys initialUtxo)
+
+      -- UTxO in the genesis block for the rich men
+      initialUtxo :: Map Address (CC.UTxO.TxIn, CC.UTxO.TxOut)
+      initialUtxo =
+            Map.fromList
+          . mapMaybe (\(inp, out) -> mkEntry inp genRichAddr <$> keyMatchesUTxO vk out)
+          . fromCompactTxInTxOutList
+          . Map.toList
+          . CC.UTxO.unUTxO
+          . CC.UTxO.genesisUtxo
+          $ gc
+        where
+          mkEntry :: CC.UTxO.TxIn
+                  -> Address
+                  -> CC.UTxO.TxOut
+                  -> (Address, (CC.UTxO.TxIn, CC.UTxO.TxOut))
+          mkEntry inp addr out = (addr, (inp, out))
+
+      keyMatchesUTxO :: Crypto.VerificationKey -> CC.UTxO.TxOut -> Maybe CC.UTxO.TxOut
+      keyMatchesUTxO key out =
+        if CC.Common.checkVerKeyAddress key (CC.UTxO.txOutAddress out)
+        then Just out else Nothing
+
+      fromCompactTxInTxOutList :: [(CC.UTxO.CompactTxIn, CC.UTxO.CompactTxOut)]
+                               -> [(CC.UTxO.TxIn, CC.UTxO.TxOut)]
+      fromCompactTxInTxOutList =
+          map (bimap CC.UTxO.fromCompactTxIn CC.UTxO.fromCompactTxOut)
+      cheat     :: CC.UTxO.TxWitness
```

Comment:
Unrelated to this PR but, at some point, we should probably come up with a better name for this.

Reply:
Agree with you :100:

Reply:
I actually fixed this in https://github.com/input-output-hk/cardano-node/pull/152/files#diff-75f92b9083d544707ede537871a2a640L438

Are you sure you rebased properly, @Jimbo4350 ?

Reply:
I actually removed that name in https://github.com/input-output-hk/cardano-node/pull/152/files#diff-75f92b9083d544707ede537871a2a640L438

Are we sure we had a clean rebase, @Jimbo4350 @intricate?

Reply:
I went over the content of this PR but I'm unsure about changes in relation to #152.

Reply:
In fact, I didn't look very closely at #152. Only a couple parts where I was mentioned (and maybe some other minor things I noticed like the formatting thing).